src/overlays/actors/ovl_En_M_Thunder/z_en_m_thunder.c

```diff
@@ -29,39 +39,533 @@ const ActorInit En_M_Thunder_InitVars = {
     (ActorFunc)EnMThunder_Draw,
 };
 
-// static ColliderCylinderInit sCylinderInit = {
-static ColliderCylinderInit D_808B7120 = {
-    { COLTYPE_NONE, AT_ON | AT_TYPE_PLAYER, AC_NONE, OC1_NONE, OC2_TYPE_1, COLSHAPE_CYLINDER, },
-    { ELEMTYPE_UNK2, { 0x01000000, 0x00, 0x00 }, { 0xF7CFFFFF, 0x00, 0x00 }, TOUCH_ON | TOUCH_SFX_NONE, BUMP_ON, OCELEM_ON, },
+static ColliderCylinderInit sCylinderInit = {
+    {
+        COLTYPE_NONE,
+        AT_ON | AT_TYPE_PLAYER,
+        AC_NONE,
+        OC1_NONE,
+        OC2_TYPE_1,
+        COLSHAPE_CYLINDER,
+    },
+    {
+        ELEMTYPE_UNK2,
+        { 0x01000000, 0x00, 0x00 },
+        { 0xF7CFFFFF, 0x00, 0x00 },
+        TOUCH_ON | TOUCH_SFX_NONE,
+        BUMP_ON,
+        OCELEM_ON,
+    },
     { 200, 200, 0, { 0, 0, 0 } },
 };
 
-#endif
+static u8 sDamages[] = {
+    /* Regular    */ 1, 2, 3, 4,
+    /* Great Spin */ 1, 2, 3, 4,
+};
+
+static u16 sChargingSfxIds[] = {
+    /* ENMTHUNDER_SUBTYPE_SPIN_GREAT        */ NA_SE_IT_ROLLING_CUT_LV2,
+    /* ENMTHUNDER_SUBTYPE_SPIN_REGULAR      */ NA_SE_IT_ROLLING_CUT_LV1,
+    /* ENMTHUNDER_SUBTYPE_SWORDBEAM_GREAT   */ NA_SE_IT_ROLLING_CUT_LV2,
+    /* ENMTHUNDER_SUBTYPE_SWORDBEAM_REGULAR */ NA_SE_IT_ROLLING_CUT_LV1,
+};
+
+static f32 sScales[] = { 0.1f, 0.15f, 0.2f, 0.25f, 0.3f, 0.25f, 0.2f, 0.15f, 0.0f };
+
+typedef enum {
+    /* 0 */ ENMTHUNDER_SUBTYPE_SPIN_GREAT,
+    /* 1 */ ENMTHUNDER_SUBTYPE_SPIN_REGULAR,
+    /* 2 */ ENMTHUNDER_SUBTYPE_SWORDBEAM_GREAT,
+    /* 3 */ ENMTHUNDER_SUBTYPE_SWORDBEAM_REGULAR
+} EnMThunderSubType;
+
+void func_808B53C0(EnMThunder* this, PlayState* play) {
+    Player* player = GET_PLAYER(play);
+
+    this->actor.update = func_808B65BC;
+    this->isCharging = false;
+    this->subtype = ENMTHUNDER_SUBTYPE_SPIN_REGULAR;
+    this->scaleTarget = 2;
+    this->actionFunc = func_808B6310;
+    this->timer = 8;
+    this->lightColorFrac = 1.0f;
+    AudioSfx_PlaySfx(NA_SE_IT_ROLLING_CUT_LV1, &player->actor.projectedPos, 4, &gSfxDefaultFreqAndVolScale,
+                     &gSfxDefaultFreqAndVolScale, &gSfxDefaultReverb);
+    this->actor.child = NULL;
+}
+
+void EnMThunder_Init(Actor* thisx, PlayState* play) {
+    s32 pad;
+    EnMThunder* this = THIS;
+    Player* player = GET_PLAYER(play);
+
+    Collider_InitCylinder(play, &this->collider);
+    Collider_SetCylinder(play, &this->collider, &this->actor, &sCylinderInit);
+    this->type = ENMTHUNDER_GET_TYPE(&this->actor);
+    Lights_PointNoGlowSetInfo(&this->lightInfo, this->actor.world.pos.x, this->actor.world.pos.y,
+                              this->actor.world.pos.z, 255, 255, 255, 0);
+    this->lightNode = LightContext_InsertLight(play, &play->lightCtx, &this->lightInfo);
+
+    if (this->type == ENMTHUNDER_TYPE_UNK) {
+        func_808B53C0(this, play);
```

Comment:
I'd name the stuff the unktype uses as something like `EnMThunder_Unk_Setup`/``Action`/`Update` since at least then it's clear what they're for.